staticconf/config.py

```diff
@@ -439,6 +439,31 @@ def has_changed(self):
         return last_mtime < self.last_max_mtime
 
 
+def create_LoggingMTimeComparator(err_logger):
+    """Return an MTimeComparator subclass that calls err_logger(filename)
+    if os.path.getmtime(filename) fails. err_logger is called within the
+    context of the raised OSError, thus sys.exc_info() can be used to
+    examine the exception."""
```

Comment:
I think there is an easier way to accomplish this.

Modify `MTimeComparator`:

```python

def __init__(self, filenames, compare_func=None):
    self.compare_func = compare_func or os.path.getmtime
```

Then you could implement the rest of this change outside of this repo:

```python

def build_logging_mtime_comparator(err_logger):
    def func(filenames):
        return MTimeComparator(filenames, compare_func=try_getmtime_else_log)
   
    return func